### DISCUSSION TITLE
Scripting - Dynamic IP for the Meter

### DISCUSSION THREAD

**Discussion by DiegoB80** [arendst/Tasmota#21419] [Support request] [answered]
Good morning everyone,

With this very simple script, I would like to allow the user to set the IP address of the inverter to connect to via WEB UI.
I have compiled the firmware with these definitions:
#ifndef SML_REPLACE_VARS
#define SML_REPLACE_VARS
#endif
#define USE_SCRIPT_WEB_DISPLAY

If I insert the IP address of the inverter in the Meter line, everything works correctly ( "+1,[192.168.1.59],m,0,6607,modbus,1,2,r000.........")
If I replace the IP with the variable %ip%, the script doesn't display any message on the console and it doesn't work.

To be sure, I also checked via console the value of ip 
11:34:50.538 CMD: script? ip
11:34:50.540 RSL: RESULT = {"script":{"ip":"[192.168.1.59]"}}

"What am I doing wrong?"
Maybe SML_REPLACE_VARS doesn't work on descriptor line? If so, is there a work around?

Thanks

THE SCRIPT:

```
>D 
p:ip="[192.168.1.59]"

>B 
->sensor53 r 

>S
if chg[ip]>0 {
svars
}

>M 1 
+1,%ip%,m,0,6607,modbus,1,2,r00037D100004,00037D00,00037D08,00037D40,r00037D450003,r00037D48000A,00037D55,00037D57,00037D59,00037D6A,000390FF,000390F9,00039101,00037D72,000390CA
1,000308SSss@i0:10,PV1 Voltage,V,PV1_Voltage,1 
1,000308x2SSss@i0:100,PV1 Current,A,PV1_Current,2 
1,000308x4SSss@i0:10,PV2 Voltage,V,PV2_Voltage,1 
1,000308x6SSss@i0:100,PV2 Current,A,PV2_Current,2 
1,000304UUuu@i1:1,State Code,SC,State_Code,0 
1,000304UUuuUUuu@i2:1,Error Code,EC,Error_Code,0 
1,000304SSssSSss@i3:1,Input Power,W,Input_Power,0 
1,000306UUuu@i4:10,Phase 1 Voltage,V,Phase_1_Voltage,1 
1,000306x2UUuu@i4:10,Phase 2 Voltage,V,Phase_2_Voltage,1 
1,000306x4UUuu@i4:10,Phase 3 Voltage,V,Phase_3_Voltage,1 
1,000314SSssSSss@i5:1000,Phase 1 Current,A,Phase_1_Current,2 
1,000314x4SSssSSss@i5:1000,Phase 2 Current,A,Phase_2_Current,2 
1,000314x8SSssSSss@i5:1000,Phase 3 Current,A,Phase_3_Current,2 
1,000314x16SSssSSss@i5:1,Active Power,W,Active_Power,0 
1,000304UUuu@i6:100,Frequency,Hz,Frequency,2 
1,000304SSss@i7:10,Internal Temperature,c,Internal_Temperature,1 
1,000304UUuu@i8:1,Status Code,SC,Status_Code,0 
1,000304UUuuUUuu@i9:100,TotaleEnergiaFV,KWh,TotaleEnergiaFV,2
1,000304UUuuUUuu@i10:100,TotaleEnergiaVenduta,KWh,TotaleEnergiaVenduta,2 
1,000304SSssSSss@i11:1,PotenzaVersoRete,W,PotenzaVersoRete,0
1,000304SSssSSss@i12:100,TotaleEnergiaPrelevata,KWh,TotaleEnergiaPrelevata,0
1,000304SSssSSss@i13:100,EnergiaProdottaOggi,KWh,EnergiaProdottaOggi,0
1,000304UUuuUUuu@i14:100,EnergiaBatteria1,KWh,EnergiaBatteria1,0
#

>W
tx(ip "Inverter IP " (150) )
```

**Comment by gemu2015** [comment, marked as answer]
this is currently not supported. i added this now but as i am very busy with other things i have no time to verify it .
replace the source and report back if everything works now

[xsns_53_sml.ino.zip](https://github.com/arendst/Tasmota/files/15335108/xsns_53_sml.ino.zip)

**Reply by DiegoB80** [reply]
Thank you very much for the very fast response.
I will test it today and let you know!

**Reply by DiegoB80** [reply]
Hi Gemu2015,

I got this error:
00:00:22.979 BRY: failed to run compiled code (key_error - modbus)

If it's possible when you check the code, enable the possibility to have also the "port" field dynamic.
So also this line becomes valid:
+1,%ip%,m,0,%port%,modbus,1,2,r00037D100004,00037D00,00037D08,00037D40,r00037D450003,r00037D4800

Thank you very much!

**Reply by sfromis** [reply]
`00:00:22.979 BRY: failed to run compiled code (key_error - modbus)`

Is based on the Berry code you're running, typically when trying to get data from a map using the key `modbus`, which was not in the payload it is trying to parse.

**Reply by DiegoB80** [reply]
Sorry,

my fault! It works!

Congratulations on the support, so fast and efficient.

**Reply by DiegoB80** [reply]
Sorry @gemu2015 ,

can you update the Tasmota repository with your changes (xsns_53_sml.ino.zip)?
Currently, if I compile new tasmota releases, I lose "Dynamic IP " feature.

Thanks

**Reply by gemu2015** [reply]
hat was already merged some time ago!

**Reply by DiegoB80** [reply]
Confirmed. Works with last build 14.1.0.4 
Thanks
